orangecontrib/timeseries/timeseries.py

```diff
@@ -22,13 +22,23 @@ def __init__(self, time_values):
         self.time_values = time_values
         self.backwards_compatible_delta = self._get_backwards_compatible_delta()
 
-        if len(time_values) <= 1:
+        def no_delta():
             self.deltas = []
             self.is_equispaced = True
             self.min = None
+
+        if len(time_values) <= 1:
+            no_delta()
             return
 
-        deltas = list(np.sort(np.unique(np.diff(self.time_values))))
+        deltas = list(np.sort(np.unique(np.diff(np.sort(self.time_values)))))
+
+        # in case several rows fall on the same datetime, remove the zero
+        if deltas[0] == 0:
+            if len(deltas) == 1:
+                no_delta()
+                return
+            deltas = deltas[1:]
```

Comment:
`deltas = deltas[1:] `could also result in `len(deltas) == 1`, but wouldn't be checked.

```
deltas = list(np.sort(np.unique(np.diff(np.sort(self.time_values)))))
if deltas and deltas[0] == 0:
    deltas.pop(0)
if len(deltas) <= 1:
    no_deltas()
```

I think it should do the same thing, but cover all the cases.